src/main/java/com/minecolonies/coremod/entity/ai/citizen/undertaker/EntityAIWorkUndertaker.java

```diff
@@ -347,6 +347,7 @@ private IAIState tryResurrect()
             if (chance >= random.nextDouble())
             {
                 final ICitizenData citizenData = buildingGraveyard.getColony().getCitizenManager().resurrectCivilianData(buildingGraveyard.getLastGraveData().getCitizenDataNBT(), true, world, gravePos);
+                citizenData.setWorkBuilding(null);
```

Comment:
Reading my own log, I thought I saw the error occured in the resurrectCivilianData call in the line above.
If so, this won't fix anything.

Reply:
True, I thought this was on the next AI tick....